code/datums/components/anti_magic.dm

```diff
@@ -15,12 +15,43 @@
 	holy = _holy
 
 /datum/component/anti_magic/proc/on_equip(datum/source, mob/equipper, slot)
+	if(slot == SLOT_IN_BACKPACK)
+		return
 	RegisterSignal(equipper, COMSIG_MOB_RECEIVE_MAGIC, .proc/can_protect, TRUE)
 
 /datum/component/anti_magic/proc/on_drop(datum/source, mob/user)
 	UnregisterSignal(user, COMSIG_MOB_RECEIVE_MAGIC)
 
-/datum/component/anti_magic/proc/can_protect(datum/source, _magic, _holy, list/protection_sources)
+/datum/component/anti_magic/proc/can_protect(datum/source, _magic, _holy, major, list/protection_sources)
 	if((_magic && magic) || (_holy && holy))
 		protection_sources += parent
+		react(major)
```

Comment:
Have an optional react callback instead of something on the component

Reply:
how would i achieve this?